Experiments/ExperimentsTests/VariationCacheTests.swift

```diff
@@ -0,0 +1,36 @@
+import XCTest
+@testable import Experiments
+
+final class VariationCacheTests: XCTestCase {
+    func test_variation_is_nil_when_the_value_does_not_exist() throws {
+        // Given
+        let userDefaults = try XCTUnwrap(UserDefaults(suiteName: UUID().uuidString))
+
+        // When
+        let cache = VariationCache(userDefaults: userDefaults)
+
+        // Then
+        XCTAssertNil(cache.variation(for: .applicationPasswordAuthentication))
+    }
+
+    func test_correct_variation_is_returned_after_setting_it() throws {
+        // Given
+        let userDefaults = try XCTUnwrap(UserDefaults(suiteName: UUID().uuidString))
+        let cache = VariationCache(userDefaults: userDefaults)
+
+        // When
+        try cache.assign(variation: .treatment, for: .applicationPasswordAuthentication)
+
+        // Then
+        XCTAssertEqual(cache.variation(for: .applicationPasswordAuthentication), .treatment)
+    }
+
+    func test_it_throws_when_trying_to_cache_logged_in_experiment() throws {
+        // Given
+        let userDefaults = try XCTUnwrap(UserDefaults(suiteName: UUID().uuidString))
+        let cache = VariationCache(userDefaults: userDefaults)
+
+        // When
+        XCTAssertThrowsError(try cache.assign(variation: .treatment, for: .aaTestLoggedIn))
```

Comment:
Nit: Same comment as above about mocking A/B test.

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/8806/commits/e0d9ee5a75b8da40669fe2e7d605238f50b1b73b